pkg/driver/reaper.go

```diff
@@ -45,20 +49,25 @@ func (r *reaper) start() {
 	go r.runLoop()
 }
 
-// runLoop waits for all child processes that exit
-// currently only stunnel process is created by efs mount helper
-// and is inherited as the child process of the driver
+// runLoop catches SIGCHLD signals to remove stunnel zombie processes. stunnel
+// processes are guaranteed to become zombies because their parent is the
+// driver process but they get terminated by the efs-utils watchdog process:
+// https://github.com/aws/efs-utils/blob/v1.31.2/src/watchdog/__init__.py#L608
+// pid_max is 4194303.
 func (r *reaper) runLoop() {
 	for {
 		select {
 		case <-r.sigs:
-			var status syscall.WaitStatus
-			var rusage syscall.Rusage
-			childPid, err := syscall.Wait4(-1, &status, syscall.WNOHANG, &rusage)
+			procs, err := ps.Processes()
 			if err != nil {
-				klog.Warningf("Failed to wait for child process %s", err)
+				klog.Warningf("reaper: failed to get all procs: %v", err)
 			} else {
-				klog.V(4).Infof("Waited for child process %d", childPid)
+				for _, p := range procs {
+					reaped := waitIfZombieStunnel(p)
+					if reaped {
```

Comment:
This will stop after encountering first zombie process. What happens if there are more?

Reply:
each SIGCHLD received from the channel should result in just one process  being reaped. It doesn't matter if the process we are reaping is the same process that got terminated and resulted in us getting a SIGCHLD as long as we always clean up one zombie per SIGCHLD. I'll add a comment that reaping one process per signal received over channel is intentional